File: Client Applications/Python Client/album_search.py
import util
import json
import urllib
import settings

def beginAlbumSearch(albumTitle, applicationID, applicationKey):

	# Set the request URL
	url = settings.apiAddress + 'Albums/?albumTitle=' + urllib.parse.quote_plus(albumTitle) + '&format=json'
		
	# Send the GET request
	resp = util.request(url, applicationID, applicationKey)

	# Interpret the JSON response 
	data = json.loads(resp.decode('utf8'))

	# Return the collection of album objects
	return data['ResultSet']

# Search for albums by title	
albumTitle = input('Please enter a album title:')
result = beginAlbumSearch(albumTitle, settings.applicationID, settings.applicationKey)

# Output the search results
for album in result:
	print(album['Name'] + ' - ' + album['Artists'])

input('Press Enter to continue...')

File: Client Applications/Ruby Client/album_search.rb
require './util'
require './settings'
require 'open-uri'
require 'rubygems'
require 'json'

def beginAlbumSearch(albumTitle, applicationID, applicationKey)

	# Set the request URL
	url = $apiAddress + 'Albums/?albumTitle=' + URI::encode(albumTitle) + '&format=json'
		
	# Send the GET request
	resp = request(url, applicationID, applicationKey)

	# Interpret the JSON response 
	data = JSON.parse(resp.body)

	# Return the collection of album objects
	return data['ResultSet']
	
end

# Search for albums by title	
puts 'Please enter a album title:'
albumTitle = gets.chomp
result = beginAlbumSearch(albumTitle, $applicationID, $applicationKey)

# Output the search results
result.each() { 
	|album| puts album['Name'] + ' - ' + album['Artists']
}

puts 'Press Enter to continue...'
input = gets.chomp